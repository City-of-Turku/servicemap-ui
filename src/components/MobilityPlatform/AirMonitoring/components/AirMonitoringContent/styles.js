const styles = (theme) => ({
  popupInner: {
    borderRadius: '3px',
    marginBottom: '0.4rem',
    marginLeft: '0.6rem',
    lineHeight: 1.2,
    overflowX: 'hidden',
  },
  buttonTransparent: {
    backgroundColor: '#fff',
    border: 'none',
    cursor: 'pointer',
    '&:hover': {
      color: 'rgba(84, 84, 84, 255)',
    },
  },
  button: {
    border: '1px solid gray',
    borderRadius: '5px',
    cursor: 'pointer',
    marginRight: theme.spacing(1.5),
  },
  buttonGray: {
    backgroundColor: '#ddd',
    border: 'none',
    borderRadius: '5px',
    color: '#fff',
    cursor: 'pointer',
    padding: theme.spacing(1),
  },
  buttonWhite: {
    backgroundColor: '#fff',
  },
  buttonActive: {
    backgroundColor: 'rgba(7, 44, 115, 255)',
    color: '#fff',
  },
  paddingWide: {
    padding: `${theme.spacing(0.5)} ${theme.spacing(1)}`,
  },
  paddingNarrow: {
    padding: theme.spacing(0.5),
  },
  widthMd: {
    width: '95%',
  },
  widthSm: {
    width: '87%',
  },
  contentHeader: {
    display: 'flex',
    flexDirection: 'row',
    marginTop: theme.spacing(0.5),
    marginBottom: theme.spacing(1.5),
    alignItems: 'flex-end',
    borderBottom: '2px solid gray',
    justifyContent: 'space-between',
  },
  headerSubtitle: {
    padding: '4px 0 5px',
    fontWeight: 'bold',
    marginBlockStart: theme.spacing(2),
    marginBlockEnd: theme.spacing(0.2),
  },
  dateContainer: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    maxWidth: '32%',
  },
  container: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
  },
  parameterTypes: {
    display: 'flex',
    justifyContent: 'space-evenly',
    alignItems: 'center',
    marginBottom: theme.spacing(0.75),
  },
  parameterTypeText: {
    fontWeight: 'bold',
    fontSize: '0.8rem',
    paddingTop: theme.spacing(0.5),
    paddingRight: theme.spacing(1),
  },
  buttonText: {
    fontSize: '0.75rem',
  },
  chartContainer: {
    margin: 0,
  },
  iconWrapper: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'rgba(7, 44, 115, 255)',
    color: '#fff',
    border: '1px solid gray',
    borderRadius: '5px',
    padding: theme.spacing(0.5),
    width: '30px',
    height: '30px',
  },
  dateStepsContainer: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'space-evenly',
    alignItems: 'center',
    padding: '1rem 0',
  },
});

export default styles;
