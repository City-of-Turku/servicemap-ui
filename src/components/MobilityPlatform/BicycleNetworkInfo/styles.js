const styles = {
  container: {
    margin: '0.5rem 1rem',
  },
  routeInfo: {
    border: '1px solid #000000',
    borderRadius: '5px',
    padding: '0.6rem',
    marginTop: '0.5rem',
  },
};

export default styles;
