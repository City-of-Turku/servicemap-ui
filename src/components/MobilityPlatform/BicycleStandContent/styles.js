const styles = {
  title: {
    marginBottom: '0.5rem',
    width: '85%',
    borderBottom: '1px solid #000000',
  },
};

export default styles;
