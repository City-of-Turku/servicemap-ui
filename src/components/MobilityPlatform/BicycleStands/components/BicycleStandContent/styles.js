export default theme => ({
  title: {
    marginBottom: theme.spacing(1),
    width: '85%',
    borderBottom: '1px solid #000000',
  },
  titleText: {
    fontSize: '0.9rem',
  },
  paragraph: {
    marginBottom: theme.spacing(0.5),
  },
  padding: {
    padding: theme.spacing(1),
  },
});
