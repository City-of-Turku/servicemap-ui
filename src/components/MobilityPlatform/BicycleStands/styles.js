const styles = {
  popupInner: {
    margin: '0.8rem',
  },
};

export default styles;
