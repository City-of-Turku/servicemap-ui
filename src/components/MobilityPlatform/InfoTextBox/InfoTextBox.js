import { Link, Typography } from '@material-ui/core';
import PropTypes from 'prop-types';
import React from 'react';

const InfoTextBox = ({
  classes, intl, infoText, linkUrl, linkText, reducePadding,
}) => (
  <div className={`${classes.container} ${reducePadding ? classes.paddingSm : classes.padding}`}>
    <Typography
      variant="body2"
      aria-label={intl.formatMessage({
        id: infoText,
      })}
    >
      {intl.formatMessage({
        id: infoText,
      })}
    </Typography>
    {linkUrl ? (
      <Link target="_blank" href={linkUrl}>
        <Typography className={classes.link} variant="body2" aria-label={linkUrl}>
          {intl.formatMessage({
            id: linkText,
          })}
        </Typography>
      </Link>
    ) : null}
  </div>
);

InfoTextBox.propTypes = {
  classes: PropTypes.objectOf(PropTypes.any).isRequired,
  intl: PropTypes.objectOf(PropTypes.any).isRequired,
  infoText: PropTypes.string,
  linkUrl: PropTypes.string,
  linkText: PropTypes.string,
  reducePadding: PropTypes.bool,
};

InfoTextBox.defaultProps = {
  infoText: '',
  linkUrl: '',
  linkText: '',
  reducePadding: false,
};

export default InfoTextBox;
