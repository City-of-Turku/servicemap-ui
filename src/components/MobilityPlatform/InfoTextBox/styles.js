export default theme => ({
  container: {
    padding: theme.spacing(2),
    textAlign: 'left',
    paddingBottom: theme.spacing(2),
  },
  link: {
    marginTop: theme.spacing(0.5),
    color: theme.palette.link.main,
    textDecoration: 'underline',
  },
});
