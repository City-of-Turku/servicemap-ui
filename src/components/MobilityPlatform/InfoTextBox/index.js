import { withStyles } from '@material-ui/core';
import { injectIntl } from 'react-intl';
import InfoTextBox from './InfoTextBox';
import styles from './styles';

export default withStyles(styles)(injectIntl(InfoTextBox));
