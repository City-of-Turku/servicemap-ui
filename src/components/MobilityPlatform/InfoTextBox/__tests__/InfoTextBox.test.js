// Link.react.test.js
import React from 'react';
import { ThemeProvider } from '@mui/material/styles';
import { render } from '@testing-library/react';
import { IntlProvider } from 'react-intl';
import themes from '../../../../themes';
import InfoTextBox from '../index';
import finnishTranslations from '../../../../i18n/fi';

// Mock props for intl provider
const intlMock = {
  locale: 'fi',
  messages: finnishTranslations,
};

const mockProps = {
  showChargingStations: true,
  infoText: 'mobilityPlatform.info.chargingStations',
  linkUrl: '',
  linkText: '',
};

// eslint-disable-next-line react/prop-types
const Providers = ({ children }) => (
  <IntlProvider {...intlMock}>
    <ThemeProvider theme={themes.SMTheme}>{children}</ThemeProvider>
  </IntlProvider>
);

const renderWithProviders = component => render(component, { wrapper: Providers });

describe('<InfoTextBox />', () => {
  it('should work', () => {
    const { container } = renderWithProviders(<InfoTextBox {...mockProps} />);
    expect(container).toMatchSnapshot();
  });

  it('does show text correctly', () => {
    const { container } = renderWithProviders(<InfoTextBox {...mockProps} />);

    const p = container.querySelectorAll('p');
    expect(p[0].textContent).toContain(finnishTranslations['mobilityPlatform.info.chargingStations']);
  });

  it('does contain aria-label attribute', () => {
    const { container } = renderWithProviders(<InfoTextBox {...mockProps} />);

    const p = container.querySelectorAll('p');
    expect(p[0].getAttribute('aria-label')).toContain(finnishTranslations['mobilityPlatform.info.chargingStations']);
  });
});
