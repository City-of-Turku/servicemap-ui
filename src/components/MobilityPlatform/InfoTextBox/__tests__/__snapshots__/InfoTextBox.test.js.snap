// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InfoTextBox /> should work 1`] = `
<div>
  <div
    class="injectIntl(InfoTextBox)-container-1 injectIntl(InfoTextBox)-padding-3"
  >
    <p
      aria-label="Turun alueen julkiset autojen sähkölatauspisteet. Latauspistetiedot perustuvat 05/2022 tehtyyn kartoitukseen."
      class="MuiTypography-root MuiTypography-body2 css-1ak20dk-MuiTypography-root"
    >
      Turun alueen julkiset autojen sähkölatauspisteet. Latauspistetiedot perustuvat 05/2022 tehtyyn kartoitukseen.
    </p>
  </div>
</div>
`;
